File: src/org/cerion/symcalc/expression/function/list/First.java
package org.cerion.symcalc.expression.function.list;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.ListExpr;
import org.cerion.symcalc.expression.FunctionExpr;

public class First extends FunctionExpr {

	public First(Expr... e) {
		super(FunctionType.LAST,e);
	}
	
	@Override
	protected Expr evaluate() {
		
		if(get(0).isList())
		{
			ListExpr L = (ListExpr)get(0);
			return L.get(0);
		}
		
		return this;
	}
}

File: src/org/cerion/symcalc/expression/function/list/First.kt
package org.cerion.symcalc.expression.function.list

import org.cerion.symcalc.expression.Expr
import org.cerion.symcalc.expression.FunctionExpr
import org.cerion.symcalc.expression.ListExpr

class First(vararg e: Expr) : FunctionExpr(FunctionExpr.FunctionType.LAST, *e) {

    override fun evaluate(): Expr {
        if (get(0).isList) {
            return get(0).asList().get(0)
        }

        return this
    }
}
